packages/calcite-components/src/components/block/block.tsx

```diff
@@ -237,16 +237,17 @@ export class Block
   }
 
   renderIcon(): VNode[] {
-    const { el, status } = this;
+    const { el, loading, messages, status } = this;
 
     const showingLoadingStatus = this.loading && !this.open;
-
     const statusIcon = showingLoadingStatus ? ICONS.refresh : ICONS[status];
 
     const hasIcon = getSlotted(el, SLOTS.icon) || statusIcon;
 
     const iconEl = !statusIcon ? (
       <slot key="icon-slot" name={SLOTS.icon} />
+    ) : loading ? (
+      <calcite-loader inline label={messages.loading} />
     ) : (
       <calcite-icon
         class={{
```

Comment:
I think `[CSS.loading]: showingLoadingStatus`, and its corresponding `CSS` prop and styles, need to go as well since it's no longer applicable (we have the single loader now to indicate loading).